keanu-python/keanu/tensor.py

```diff
@@ -24,24 +25,23 @@ def __init__(self, t):
 
     @staticmethod
     def __get_tensor_from_ndarray(ndarray):
-        normalized_ndarray = Tensor.__ensure_rank_is_atleast_two(ndarray)
-
-        ctor = Tensor.__infer_tensor_ctor_from_ndarray(normalized_ndarray)
-        values = k.to_java_array(normalized_ndarray.flatten().tolist())
-        shape = k.to_java_long_array(normalized_ndarray.shape)
+        ctor = Tensor.__infer_tensor_ctor_from_ndarray(ndarray)
+        values = k.to_java_array(ndarray.flatten().tolist())
+        shape = k.to_java_long_array(ndarray.shape)
 
         return ctor(values, shape)
 
-    @staticmethod
-    def __ensure_rank_is_atleast_two(ndarray):
-        if len(ndarray.shape) == 1:
-            return ndarray[..., None]
-        else:
-            return ndarray
-
     @staticmethod
     def __infer_tensor_ctor_from_ndarray(ndarray):
-        if len(ndarray) == 0:
+
+        if isinstance(ndarray, bool_types):
```

Comment:
This method is only called at loc 18 and 20, so how is it possible that an ndarray is a primitive?